hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeManager.java

```diff
@@ -71,64 +86,67 @@ public FeeManager(@NonNull final ExchangeRateManager exchangeRateManager) {
-                        + "transactions.");
-            }
+        // Get the current schedule
+        var currentSchedule = schedules.currentFeeSchedule();
+        if (currentSchedule == null) {
+            // If there is no current schedule, then we default to the default schedule. Since the default
+            // schedule is completely empty, this will effectively disable the handling of any transactions,
+            // since we don't know what to charge for them.
+            logger.warn("Unable to parse current fee schedule, will default to an empty schedule, effectively"
+                    + "disabling all transactions.");
+            currentSchedule = FeeSchedule.DEFAULT;
+        }
 
-            // Get the expiration time of the current schedule
-            if (currentSchedule.hasExpiryTime()) {
-                this.currentScheduleExpirationSeconds =
-                        currentSchedule.expiryTimeOrThrow().seconds();
-            } else {
-                // If we don't have an expiration time, then we default to 0, which will effectively expire the
-                // current schedule immediately. This is the safest option.
-                logger.warn("The current fee schedule has no expiry time, defaulting to 0, effectively expiring it"
-                        + "immediately");
-                this.currentScheduleExpirationSeconds = 0;
-            }
+        // Populate the map of HederaFunctionality -> FeeData for the current schedule
+        this.currentFeeDataMap = new HashMap<>();
+        if (currentSchedule.hasTransactionFeeSchedule()) {
+            populateFeeDataMap(currentFeeDataMap, currentSchedule.transactionFeeScheduleOrThrow());
+        } else {
+            logger.warn("The current fee schedule is missing transaction information, effectively disabling all"
+                    + "transactions.");
+        }
 
-            // Get the next schedule
-            var nextSchedule = schedules.nextFeeSchedule();
-            if (nextSchedule == null) {
-                // If there is no next schedule, then we default to the current schedule. If we didn't have a current
-                // schedule either, then basically we have an empty schedule with an expiration time of 0, which will
-                // still get used since we continue to use the next schedule even if the expiration time has passed.
-                logger.warn("Unable to parse next fee schedule, will default to the current fee schedule.");
-                nextFeeDataMap = new HashMap<>(currentFeeDataMap);
+        // Get the expiration time of the current schedule
+        if (currentSchedule.hasExpiryTime()) {
+            this.currentScheduleExpirationSeconds =
+                    currentSchedule.expiryTimeOrThrow().seconds();
+        } else {
+            // If we don't have an expiration time, then we default to 0, which will effectively expire the
+            // current schedule immediately. This is the safest option.
```

Comment:
It is a safer option, but does monoservice do this (and if not do we need to match monoservice for now)?  I thought monoservice defaulted to continuing to use the "next" schedule indefinitely if the expiration time was 0.

Reply:
Again, I am not that familiar with the code. I believe this is what happens here. The current schedule expires immediately and we use the next schedule until something changes.